nixos/modules/services/display-managers/gdm.nix

```
@@ -7,7 +7,7 @@
 
 let
 
-  cfg = config.services.xserver.displayManager;
+  cfg = config.services.displayManager;
```

At this point, just do:

```suggestion
  cfg = config.services.displayManager.gdm;
```

and replace uses of cfg.gdm properly.